packages/data-context/test/unit/sources/GitDataSource.spec.ts

```diff
@@ -62,28 +60,22 @@ describe('GitDataSource', () => {
       isRunMode: false,
       projectRoot: projectPath,
       onBranchChange,
-      onGitInfoChange,
-      onError,
+      onGitInfoChange: dfd.resolve,
+      onError: (err: any) => {
+        assert.fail(err)
+      },
     })
 
-    fs.createFileSync(fooSpec)
-    fs.writeFileSync(xhrSpec, 'it(\'modifies the file\', () => {})')
+    await fs.createFile(fooSpec)
+    await fs.writeFile(xhrSpec, 'it(\'modifies the file\', () => {})')
 
     gitInfo.setSpecs([fooSpec, aRecordSpec, xhrSpec])
 
-    let result: any[] = []
-
-    do {
-      result = await Promise.all([
-        gitInfo.gitInfoFor(fooSpec),
-        gitInfo.gitInfoFor(aRecordSpec),
-        gitInfo.gitInfoFor(xhrSpec),
-      ])
-
-      await new Promise((resolve) => setTimeout(resolve, 100))
-    } while (result.some((r) => r == null))
+    await dfd.promise
```

Comment:
```suggestion
    const gitInfoChangeResolve = await dfd.promise

    expect(gitInfoChangeResolve).to.eql(filepaths)
```

I did not see anywhere in the test where it was verifying what was returned from the `onGitInfoChange` callback. Do you think it is worth having this change?  This does not need to hold up this PR though since it is unrelated.

Reply:
Good suggestion 👍 , added